documentation/APIs/core-models.md

```diff
@@ -0,0 +1,581 @@
+The slot index of timestamp `ts` is `(ts - genesisTimestamp)/duration + 1`.
+
+## Slot Commitment
+
+Commitment is an object that contains a summary of a slot, and it is linked to the commitment of the previous slot, which forms a commitment chain. 
+
+Multiple sparse merkle trees are managed by per slot, which are:
+* *Tangle tree*: All accepted blocks within a slot.
+* *State mutation tree*: All accepted transactions within a slot.
+* *Activity tree*: All active accounts within a slot.
+* *State tree*: All accepted created/consumed outputs within a slot.
+* *Mana tree*: Mana of all accounts at the end of slot.
+
+When a slot is committable, new commitment is generated and will be included in the new issued blocks.
+
+**Note** that more trees will be added in the future, but this will only change the calculation of `Root ID`, the Slot Commitment structure stays the same.
+
+The following table describes the serialization of a Slot Commitment:
+
+<table>
+  <tr>
+    <th>Name</th>
+    <th>Type</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>Index</td>
+    <td>int64</td>
+    <td>
+      The slot index of this commitment. It is calculated based on genesis timestamp and the duration of a slot.
+    </td>
+  </tr>
+  <tr>
+    <td>Previous Slot Commitment ID</td>
+    <td>ByteArray[40]</td>
+    <td>The commitment ID of the previous slot.</td>
+  </tr>
+  <tr>
+    <td>Root ID</td>
+    <td>ByteArray[32]</td>
+    <td>
+      A BLAKE2b-256 hash of concating multiple sparse merkle tree roots of a slot.
+      <i>Root ID = hash( Concat( hash(Concat(Tangle, TX)), hash(Concat(State, Mana)) ) )</i>  
+    </td>
+  </tr>
+  <tr>
+    <td>Cumulative Weight</td>
+    <td>int64</td>
+    <td>The sum of previous slot commitment cumulative weight and weight of issuers of accepted blocks within this slot. It is just an indication of "committed into" this slot, and can not strictly be used for evaluating the switching of a chain.</td>
+  </tr>
+</table>
+
+### Slot Commitment ID
+
+Slot Commitment ID denotes an identifier of a slot commitment, with type `ArrayBytes[40]`. It is calculated as the following steps, using BLAKE2b-256 hash function:
+
+* `content` is the serialized slot commitment.
+* `slot_index` is the slot index of the slot commitment.
+
+Caculation:
+1. `content_hash` = hash(`content`)
+2. And finally, `CommitmentID` = Concat(`content_hash`, `slot_index`) 
+
+The string format of Slot Commitment ID is hexadecimal encoding of Slot Commitment ID with `0x` prefix.
+
+## Payloads
+The following table lists all currently specified payloads that can be part of a block. 
+
+<table>
+    <tr>
+        <th>Name</th>
+        <th>Payload Type</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td>Tagged Data</td>
+        <td>5</td>
+        <td>A payload which holds a tag and associated data.</td>
+    </tr>
+    <tr>
+        <td>Transaction</td>
+        <td>6</td>
```

Comment:
Do we not need a new payload type since this transaction is not compatible with the stardust one?

Reply:
New version is introduced in transaction essence type.

Reply:
Ah yeah we keep the transaction but bump the essence, I see. Thanks!